### PR TITLE
[RISCV] Pattern-match frameindex

### PR DESCRIPTION
**Pull request by s-barannikov**

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVInstrInfo.td [hunk 1]
```diff
@@ -1400,8 +1397,19 @@ def PseudoAddTPRel : Pseudo<(outs GPR:$rd),
...
+def to_tframeindex : SDNodeXForm<frameindex, [{
+  return CurDAG->getTargetFrameIndex(N->getIndex(), N->getValueType(0));
+}]>;
+
+def : GICustomOperandRenderer<"renderFrameIndex">,
+      GISDNodeXFormEquiv<to_tframeindex>;
+
+def : Pat<(frameindex:$fi), (ADDI (iPTR (to_tframeindex $fi)), 0)>;
+
+def : Pat<(add_like frameindex:$fi, simm12:$offset),
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVInstrInfo.td:1410]: I think add_like might be less capable than `SelectionDAG::isBaseWithConstantOffset`, but if it shows up its probably fixable.

**Review by topperc — APPROVED** [review]
LGTM

File: llvm/lib/Target/RISCV/RISCVInstrInfo.td (hunk 1 above)
Comment (s-barannikov) [reply llvm/lib/Target/RISCV/RISCVInstrInfo.td:1410]: `isBaseWithConstantOffset` additionally handles `(xor x, INT_MIN)`, that's all the difference I noticed.

**Comment by s-barannikov** [comment]
Looks like I'll have to fix the tablegen backend first.

File: llvm/lib/Target/RISCV/RISCVInstrInfo.td (hunk 1 above)
Comment (lenary) [reply llvm/lib/Target/RISCV/RISCVInstrInfo.td:1411]: Given you think there is a difference, please don't mark this as "NFCI" in the message.

Comment (s-barannikov) [reply llvm/lib/Target/RISCV/RISCVInstrInfo.td:1411]: I removed NFCI now

**Pull request merged**